### PR TITLE
Extensions for Vault Location List

### LINKED ISSUE #163: Extensions for `Vault Location List`
**Update: 26.4.2023** - for `Phase 1` of this we will keep the existing vault-location data. We'll implement `racks`, `shelves` and `filter by rack` in a future issue.

Mockup for muster form [here](https://docs.google.com/presentation/d/1eCTxSe2yML3xb3y5c9uS8zm3GdotJymJIZ-6NrfrpZg/edit#slide=id.g23a71d30cbf_0_0)

The vault consists of lots of racks, each with lots of shelves.

The racks are numbered A to G, and the shelves numbered 01 to 66. 

There are also some separate named locations, such as `A667`, `B776`.

When users are checking the contents of the shelves (doing a _muster_), they will choose a group of shelves near each other. So, it will be one or two racks, then selecting some number of shelves off those racks.

Users need these abilities:
- filter by rack
- select one or more vault locations (shelves)
- print off combined muster document(s) for those locations.  This is one printed report, but it can contain muster lists for a number of shelves, each starting on a new page.  Report formatted as below

I _think_ we can handle the above by the `List` component generating a new field from the data: "Rack".  Where the `vault-location` is in the format `A00` the `Rack` is the first letter`.  Where the vault-location isn't formatted like that, then the `Rack` field is empty.

## Report format
- one long document, made up of multiple location reports, each starting on a new page.  For each report:
    - title: `RCO - Location Muster List`
    - sub-title: `100 percent Muster List for location G12, printed 21 Apr 2023 14:14 (17 items)
    - detail: list with these columns:
        - Item Number
        - Media Type
        - Serial/Pages
        - Prot Marking
        - State
    - footer 1: summary count of above items, grouped by protective marking (see summary below)
    - footer 2: signature block, as shown below

### Footer summary block
The users need to see a count of how many items at each protective marking were present in the previous list.  Presented in boxed area as:
* Prot Marking 1: 23
* Prot Marking 2: 14

### Signature block:
image

### PR DESCRIPTION
**Pull request by salony75way**
fixed #163

### CODE REVIEW

File: src/components/VaultLocationReport.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              <Box padding={'20px'} key={id}>
+                <Typography variant='h4' textAlign='center' margin='10px'>
+                  RCO - Location Muster List
+                </Typography>
+                <Typography variant='h5' textAlign='center' margin='10px'>
+                  100% Muster List for {locations?.[id]?.name}, printed{' '}
+                  {DateTime.fromISO(new Date().toISOString()).toFormat(
+                    'dd/MMM/yyyy hh:mm'
+                  )}{' '}
+                  (17 items)
```
Comment (IanMayo) [inline src/components/VaultLocationReport.tsx:65]: Hello @salony75way - the above should be a count of the number of items at this location.

Comment (salony75way) [reply src/components/VaultLocationReport.tsx:65]: Hi @IanMayo 
yes, I am currently working on these remaining changes.

Comment (IanMayo) [reply src/components/VaultLocationReport.tsx:65]: No problem, I was just checking you hadn't missed it :-D

**Comment by IanMayo** [comment]
This is a real edge case, but we need a better "no data found" message for the report:
image

How about "No items registered at this location"?

**Comment by IanMayo** [comment]
Hey @salony75way - it's looking great :-D

**Review by IanMayo — APPROVED** [review]

**Pull request merged**